packages/utils/src/lib/filter.ts

```diff
@@ -0,0 +1,24 @@
+export function filterItemRefsBy<T extends { refs: object[] }>(
+  items: T[],
+  refFilterFn?: (item: T['refs'][number]) => boolean,
+) {
+  return filterBy(
+    items
+      .map(item => ({
+        ...item,
+        refs: filterBy(item.refs, refFilterFn),
+      }))
+      // remove item with empty refs
+      .filter(item => item.refs.length),
+  );
+}
+
+export function filterBy<T = unknown>(
```

Comment:
This function only calls `filter`. Is it necessary to have it separate from the `filterItemRefsBy`? Will it ever be called on its own instead of calling `filter`?

Reply:
i removed filterBy